clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractMembershipManager.java

```diff
@@ -443,7 +443,7 @@ public void transitionToFatal() {
         log.error("Member {} with epoch {} transitioned to fatal state", memberId, memberEpoch);
         notifyEpochChange(Optional.empty());
 
-        if (previousState == MemberState.UNSUBSCRIBED) {
+        if (previousState == MemberState.UNSUBSCRIBED && maybeCompleteLeaveInProgress()) {
```

Comment:
I think `StreamsMembershipManager` needs this change as well. cc @mjsax

Reply:
I notice some different behavior between async and stream (https://github.com/apache/kafka/pull/18551/files#r2106243432). Maybe we can open a ticket to log it instead of changing it right now. This PR can fix the actual issue in Async consumer, WDYT?